py/builtinimport.c

```diff
@@ -346,6 +347,16 @@ STATIC void evaluate_relative_import(mp_int_t level, const char **module_name, s
     *module_name_len = new_module_name_len;
 }
 
+typedef struct _nlr_jump_callback_node_unregister_module_t {
+    nlr_jump_callback_node_t callback;
+    qstr full_mod_name;
+} nlr_jump_callback_node_unregister_module_t;
+
+STATIC void unregister_module_from_nlr_jump_callback(void *ctx_in) {
+    nlr_jump_callback_node_unregister_module_t *ctx = ctx_in;
+    mp_obj_module_unregister(ctx->full_mod_name);
```

Comment:
It would reduce code size (when LTO is not used) to inline `mp_obj_module_unregister()` directly here.

Also, no need to add more to the "public" API if it's not necessary (ie no need to add a new public function `mp_obj_module_unregister`).

If it's ever needed as a separate function, it can be factored out.

Reply:
OK.  For some reason I thought `mp_obj_module_unregister` was using private data, but you're right, it can be inlined.